subsys/bluetooth/host/hci_core.c

```diff
@@ -6355,49 +6411,39 @@ int bt_recv(struct net_buf *buf)
 		return 0;
 #endif /* BT_CONN */
 	case BT_BUF_EVT:
+	{
 #if defined(CONFIG_BT_RECV_IS_RX_THREAD)
 		hci_event(buf);
 #else
-		net_buf_put(&bt_dev.rx_queue, buf);
+		struct bt_hci_evt_hdr *hdr = (void *)buf->data;
+
+		if (bt_hci_evt_is_prio(hdr->evt)) {
+			hci_event_prio(buf);
+		}
+
+		if (bt_hci_evt_is_normal(hdr->evt)) {
+			net_buf_put(&bt_dev.rx_queue, buf);
```

Comment:
Is this safe after the `net_buf` has been processed by `hci_event_prio`? i.e. has its internal state not been modified?

Reply:
It should be, because `hci_event_prio` will save the state and restore it before returning.